compiler/z/codegen/OMRCodeGenerator.cpp

```diff
@@ -583,6 +583,48 @@ OMR::Z::CodeGenerator::CodeGenerator()
       _processorInfo.disableArch(TR_S390ProcessorInfo::TR_zNext);
       }
 
+   // Randomly disable an architecture if we are in randomGen mode
+   if (comp->getOption(TR_Randomize))
+      {
+      switch (randomizer.randomInt(TR::Compiler->target.cpu.id() - TR_s370gp7))
+         {
+         case 1:
+            {
+            _processorInfo.disableArch(TR_S390ProcessorInfo::TR_z196);
+            traceMsg(comp, "RandomGen: Disabling z196 processor architecture.");
+            break;
+            }
+
+         case 2:
+            {
+            _processorInfo.disableArch(TR_S390ProcessorInfo::TR_zEC12);
+            traceMsg(comp, "RandomGen: Disabling zEC12 processor architecture.");
+            break;
+            }
+
+         case 3:
+            {
+            _processorInfo.disableArch(TR_S390ProcessorInfo::TR_z13);
+            traceMsg(comp, "RandomGen: Disabling z13 processor architecture.");
+            break;
+            }
+
+         case 3:
```

Comment:
I think this is supposed to be:
```suggestion
         case 4:
```

Reply:
Unless the cases were supposed to start at 0...